### ISSUE TITLE
Random input: start to implement new directory structure

### ISSUE THREAD

**Issue by plocket** [plocket/docassemble-cucumber#552]
From https://github.com/SuffolkLITLab/ALKiln/discussions/531#discussioncomment-2442625

### This issue

First step: Move current artifacts to new structure:
- [x] report
- [x] debug
- [x] [redundant error screenshots (that stay out in the main folder)]
- [x] scenario folder
   - [x] [report for just the scenario]
   - [x] downloaded files
   - [x] error screenshots
   - [x] manually taken screenshots
   - [x] JSON

First, for a test run, everything would be in one artifact folder, so it can all be downloaded at the same time. Name options?
- "alkiln_tests_output-" and timestamp (dashes or underscores?)
- Somehow get the name of the test run itself. Probably possible. Plus timestamp? Plus alkiln?

This will be good in general even before starting with random input and is the first step.

Discussions on what to use in the names of artifacts: https://github.com/SuffolkLITLab/ALKiln/issues/429#issuecomment-1002657179

### Long term plan and comments

Then the below structure for the folders eventually:

> Maybe the error screenshots should be in both places. This is the idea proposed here, though the names are missing a bunch of pieces to avoid a mess and show the arrangement more clearly.
> 
> - report.txt (whole report)
> - error-3pm-scenario 1 description-endingPgID.png
> - error-4pm-scenario 2 description-endingPgID.png
> - ~3pm-endingPgID-~ scenario 1 description (folder)
>    - report-endingPgID.txt (scenario only)
>    - download-3pm-file-1-pgID.pdf
>    - error-3pm-pgID.png (Same pic as "error-3pm-scenario 1 description.png". Same name seems dumb because it would have the scenario name in it too)
>    - json-3pm-pgID.json
>    - json-3pm-pgID.json
>    - scenario 1 description.feature
>    - screenshot-3pm-pgID.png
> - ~4pm-endingPgID-~ scenario 2 description (folder)
>    - report.txt (scenario only)
>    - error-4pm-pgID.png (Same as scenario 1 error pic name rationale)
>    - scenario 2 description.feature
> 
> "3pm" and "4pm" just indicates a timestamp. "PgID" could be question id or screen trigger var, whichever is available, or neither if none available. Not sure which is preferred.

**Comment by plocket** [comment]
[This was a list of todo items, but it's been moved into the initial issue comment.]

**Comment by plocket** [comment]
We've decided to keep all scenario folders, even those that only have a report. Better that expected output is there.

**Comment by plocket** [comment]
Pretty timestamp has been requested. Edit: See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date

**Comment by plocket** [comment]
Part of the reason each scenario is in a different folder is for random input testing output - I'd like each random input scenario to have its own `.feature` file that the dev can copy/paste into their sources folder to replicate the test.

**Comment by plocket** [comment]
- [x] report.txt (whole report)
- [x] debug_log.txt
- [x] error_on-endingPgID-scenario 1 description-3pm.png
- [x] error_on-endingPgID-scenario 2 description-4pm.png
- [x] scenario 1 description-3pm (folder)
   - [x] report.txt (scenario only)
   - [x] error_on-pgID.png (Same pic as "error-scenario 1 description-endingPgID-3pm.png". Same name seems dumb because scenario name would be redundant) (excluding timestamp because there will only be max one error screenshot per scenario, so no need to make name unique.)
   - [x] ~download-file-1-pgID-3pm.pdf~ downloaded-file-name.pdf (whatever its name is in the interview. We can't change it, but also shouldn't change it. Documentation should say that all files of an interview should have a unique name)
   - [x] json_for-pgID-3pm.json
   - [x] json_for-pgID-3pm.json
   - [x] screenshot-custom-name-on-pgID-3pm.png
   - [x] screenshot_on-pgID-3pm.png
   - [ ] scenario 1 description.feature (random input only. maybe stretch goal of exploring cucumber objects to see if they have the info to make this file for regular tests.)
- [x] scenario 2 description-4pm (folder)
   - [x] report.txt (scenario only)
   - [x] error_on-pgID-4pm.png (Same as scenario 1 error pic name rationale)
   - [ ] scenario 2 description.feature

[Note: I'm including the timestamps in scenario folder names and the error screenshot names in case devs copy/paste a scenario and accidentally forget to change the description and the test fails on the same page. That is, I can easily imagine a situation in which the name wouldn't be unique without the timestamp.]

Edit:
Artifacts folder itself:
- [x] alkiln_tests_output-3pm UTC

Folder/files for our own local unit tests:
- [x] _alkiln_test-misc_artifacts
   - [x] unit_tests_logs-3pm UTC

**Comment by plocket** [comment]
Have to replay these changes on top of the new branch because I can't track down a weird bug that cropped up in the `Before()` of at least one of the tests after my attempt to merge.

**Comment by plocket** [comment]
Closed by #568 [Can't implement the feature file names because those files don't yet exist.]
